components/work/WorkEdit.tsx

```diff
@@ -1,18 +1,52 @@
-import { FormEvent, PureComponent } from 'react';
+import { FormEvent, MouseEventHandler, PureComponent } from 'react';
 import { Container } from 'react-bootstrap';
 import { formToJSON } from 'web-utility';
-
-import { WorkEditor } from './WorkEditor';
+import { Button, Col, Form, Row } from 'react-bootstrap';
 import activityStore from '../../models/Activity';
-import { TeamWork, TeamWorkModel } from '../../models/Team';
+import { TeamWork } from '../../models/Team';
 import { NextRouter, withRouter } from 'next/router';
+import { FileUpload } from '../FileUpload';
+const workTypes = [
+  {
+    title: '网站',
+    value: 'website',
+  },
+  {
+    title: '图片',
+    value: 'image',
+  },
+  {
+    title: '视频',
+    value: 'video',
+  },
+  {
+    title: 'Word',
+    value: 'word',
+  },
+  {
+    title: 'PowerPoint',
+    value: 'powerpoint',
+  },
+  {
+    title: 'PDF',
+    value: 'pdf',
+  },
+  {
+    title: 'ZIP',
+    value: 'zip',
+  },
+];
 class WorkEdit extends PureComponent<{
-  work: TeamWork;
+  work?: TeamWork;
   router: NextRouter | any;
 }> {
   store = activityStore
     .teamOf(this.props.router.query.name)
     .workOf(this.props.router.query.tid);
+  constructor(props: any) {
+    super(props);
+    this.state = { work: props.work || { type: 'website' } };
+  }
```

Comment:
```suggestion

  work = this.props.work || { type: 'website' };

```
1. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields
2. https://github.com/mobxjs/mobx/blob/mobx4and5/docs/refguide/observer-component.md#local-observable-state-in-class-based-components

Reply:
good